### ISSUE TITLE
HttpException: Invalid header field name

### ISSUE THREAD

**Issue by ghostgzt** [flutterchina/dio#969]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | flutter                           |      |
| Platform Version               | 1.17.2                             |      |
| Dio Version                    | 3.0.10                                 |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

请求某些url会报错，如https://media.kanbook.net/26181/306_cover.jpg
`
I/flutter (22804): DioError [DioErrorType.DEFAULT]: HttpException: Invalid header field name, with 32, uri = https://media.kanbook.net/26181/306_cover.jpg
`

**Comment by ghostgzt** [comment]
没人管么？

**Comment by funder7** [comment]
Is Dio accepting custom http headers?

**Comment by Narven** [comment]
@funder7 I was having this issue because by mistake I was passing a "non-default" header name. So I'm assuming that it does not support custom http headers.... OR... it needs to follow the standard format, which I think it is `X-<custom-name>`

**Comment by funder7** [comment]
I can tell you that, as far as I can remember, 32 is the ASCII symbol for enter key or space (I dont remember, anyway both are not accepted in header names) your request, or Api response, got that character in it, which leads to that error!

**Comment by Narven** [comment]
Yeah.. that was my case because I was trying sending from the API the token header in a bad format, like this

```
'Bearer ': '<token>'
```

Instead of:

```
'Authorization': 'Bearer <token>'
```

and the 32 I think it was because of the space after `Bearer`

**Comment by funder7** [comment]
I know mate, I thought the same about me when I discovered the problem :-D  ...the error message doesn't help much, though!
